packages/nuxt/src/app/components/nuxt-island.ts

```diff
@@ -160,6 +162,7 @@ export default defineComponent({
       // TODO: Validate response
       // $fetch handles the app.baseURL in dev
       const r = await eventFetch(withQuery(((import.meta.dev && import.meta.client) || props.source) ? url : joinURL(config.app.baseURL ?? '', url), {
+        url: route.fullPath,
```

Comment:
I think this will make it much harder to cache server islands as the hash we create will vary by path.

Could we instead pass this as part of the context via `createIslandPage` instead?

Reply:
I think we should not skip query 🤔, userss tend to relly on it within pages composables. Maybe can we pass an object like
```ts
{
  path: string,
query?: Record<string, string>
}
```
?

Reply:
The issue is (for example) that we will make a separate request for every query variation, which wouldn't be compatible with static sites and allow cache-busting by passing arbitrary query params or hashes ... (We should definitely not pass hash.) We could document this as a limitation of server pages... Or at a minimum, skip passing the full route if the route matches prerender routeRule?

Reply:
I see, let's start by passing only the path in the context then 🤔